src/display/annotation_layer.js

```diff
@@ -442,7 +442,7 @@ var TextWidgetAnnotationElement = (
 
   function TextWidgetAnnotationElement(parameters) {
     var isRenderable = parameters.renderInteractiveForms ||
-      (!parameters.data.hasAppearance && !!parameters.data.fieldValue);
+      (!parameters.data.appearance && !!parameters.data.fieldValue);
```

Comment:
Is `parameters.data.appearance` ever going to be defined?
Note that https://github.com/mozilla/pdf.js/pull/8065/files#diff-18ca06e2bd5be4c9132b51f78388b8f0L207 doesn't have an equivalent in the new code.